src/viam/sdk/robot/client.hpp

```diff
@@ -68,10 +119,10 @@ class RobotClient {
     static std::shared_ptr<RobotClient> with_channel(std::shared_ptr<ViamChannel> channel,
                                                      Options options);
     RobotClient(std::shared_ptr<ViamChannel> channel);
-    std::vector<ResourceName>* resource_names();
+    std::vector<Name>* resource_names();
```

Comment:
Whoa. This returns a pointer to an internal field? That seems really dangerous. How bad would it be to convert this to `const std::vector<Name>& resource_names() const`? Or even return a copy?

Reply:
Yikes, good catch! I think returning a `const&` or copy is totally fine. In my mind `const&` seems better because I'm not sure what value a copy provides to justify the extra cost, but I'm open to being convinced.

Reply:
See the other thread about this on the implementation but it definitely needs to be a copy.

Reply:
@purplenicole730 per above comment, we should definitely update this to not return a pointer.